tests/rust_syscall_tests/src/lib.rs

```diff
@@ -0,0 +1,516 @@
+// SPDX-License-Identifier: Apache-2.0
+#![no_std]
+
+pub use sallyport::libc;
+
+use core::arch::asm;
+
+pub type Result<T> = core::result::Result<T, i32>;
+
+#[macro_export]
+macro_rules! startup {
+    () => {
+        #[no_mangle]
+        #[naked]
+        pub unsafe extern "sysv64" fn _start() -> ! {
+            unsafe fn start_c(dynv: *const u64, sp: *const usize) -> ! {
```

Comment:
Can we get a comment explaining what is actually happening in this function and why it's necessary?

Reply:
yes, will add

Reply:
Tried to explain it.. Please ask, if unclear and I'll extend

Reply:
Looks good.